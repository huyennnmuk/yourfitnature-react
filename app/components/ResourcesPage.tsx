"use client";
import React from 'react';
import { track } from '../../lib/analytics';

const ResourcesPage = () => {
  const handleAffiliateClick = (link: string) => {
    track('Click', { category: 'Affiliate', label: link });
  };

  return (
    <div className="p-6 max-w-4xl mx-auto">
      <h1 className="text-4xl font-bold mb-6">Resources</h1>
      <p className="mb-6">Here are some recommended products and services to help you on your bloat-free journey. (Note: Some of these are affiliate links).</p>
      <div className="flex flex-col gap-8">
        <div className="p-6 rounded-lg">
          <h2 className="text-2xl font-bold mb-4">Meal Delivery Services</h2>
          <ul className="list-disc list-inside space-y-2">
            <li><a href="#" onClick={() => handleAffiliateClick('Green Chef')} className="text-blue-500 hover:underline" target="_blank" rel="noopener noreferrer">Green Chef</a></li>
            <li><a href="#" onClick={() => handleAffiliateClick('HelloFresh')} className="text-blue-500 hover:underline" target="_blank" rel="noopener noreferrer">HelloFresh</a></li>
            <li><a href="#" onClick={() => handleAffiliateClick('Daily Harvest')} className="text-blue-500 hover:underline" target="_blank" rel="noopener noreferrer">Daily Harvest</a></li>
          </ul>
        </div>
        <div className="p-6 rounded-lg">
          <h2 className="text-2xl font-bold mb-4">Specialty Foods</h2>
          <ul className="list-disc list-inside space-y-2">
            <li><a href="#" onClick={() => handleAffiliateClick('Thrive Market')} className="text-blue-500 hover:underline" target="_blank" rel="noopener noreferrer">Thrive Market</a></li>
            <li><a href="#" onClick={() => handleAffiliateClick('Imperfect Foods')} className="text-blue-500 hover:underline" target="_blank" rel="noopener noreferrer">Imperfect Foods</a></li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default ResourcesPage;