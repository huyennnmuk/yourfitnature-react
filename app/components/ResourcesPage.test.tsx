import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ResourcesPage from './ResourcesPage';

// Mock the analytics module
jest.mock('../../lib/analytics', () => ({
  track: jest.fn()
}));

describe('ResourcesPage', () => {
  it('renders the component', () => {
    render(<ResourcesPage />);
    expect(screen.getByText('Resources')).toBeInTheDocument();
  });

  it('renders both resource sections', () => {
    render(<ResourcesPage />);
    expect(screen.getByText('Meal Delivery Services')).toBeInTheDocument();
    expect(screen.getByText('Specialty Foods')).toBeInTheDocument();
  });

  it('has external links with proper attributes', () => {
    render(<ResourcesPage />);
    const links = screen.getAllByRole('link');
    links.forEach(link => {
      expect(link).toHaveAttribute('target', '_blank');
      expect(link).toHaveAttribute('rel', 'noopener noreferrer');
    });
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<ResourcesPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has the correct container classes for single-column layout', () => {
    const { container } = render(<ResourcesPage />);
    expect(container.firstChild).toHaveClass('p-6 max-w-4xl mx-auto');
  });
});
