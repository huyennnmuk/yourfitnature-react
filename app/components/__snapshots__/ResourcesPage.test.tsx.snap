// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ResourcesPage matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-6 max-w-4xl mx-auto"
  >
    <h1
      class="text-4xl font-bold mb-6"
    >
      Resources
    </h1>
    <p
      class="mb-6"
    >
      Here are some recommended products and services to help you on your bloat-free journey. (Note: Some of these are affiliate links).
    </p>
    <div
      class="flex flex-col gap-8"
    >
      <div
        class="p-6 rounded-lg"
      >
        <h2
          class="text-2xl font-bold mb-4"
        >
          Meal Delivery Services
        </h2>
        <ul
          class="list-disc list-inside space-y-2"
        >
          <li>
            <a
              class="text-blue-500 hover:underline"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              Green Chef
            </a>
          </li>
          <li>
            <a
              class="text-blue-500 hover:underline"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              HelloFresh
            </a>
          </li>
          <li>
            <a
              class="text-blue-500 hover:underline"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              Daily Harvest
            </a>
          </li>
        </ul>
      </div>
      <div
        class="p-6 rounded-lg"
      >
        <h2
          class="text-2xl font-bold mb-4"
        >
          Specialty Foods
        </h2>
        <ul
          class="list-disc list-inside space-y-2"
        >
          <li>
            <a
              class="text-blue-500 hover:underline"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              Thrive Market
            </a>
          </li>
          <li>
            <a
              class="text-blue-500 hover:underline"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              Imperfect Foods
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
